clients/goredis.go

```diff
@@ -1,16 +1,20 @@
 package clients
 
 import (
+	"context"
 	"fmt"
-	goredis "github.com/go-redis/redis/v7"
-	"github.com/nitishm/go-rejson/rjs"
 	"strings"
+
+	goredis "github.com/go-redis/redis/v8"
+	"github.com/nitishm/go-rejson/rjs"
 )
 
+var ctx = context.Background()
+
 // GoRedis implements ReJSON interface for Go-Redis/Redis Redis client
 // Link: https://github.com/go-redis/redis
 type GoRedis struct {
-	Conn *goredis.Client // import goredis "github.com/go-redis/redis/v7"
+	Conn *goredis.Client // import goredis "github.com/go-redis/redis/v8"
 }
```

Comment:
What if we add a `ctx context.Context` field in `GoRedis` struct and provide a method in `Handler` to change the context and return a new `Handler` object with a new `GoRedis` object. 
This way we can use different contexts in the same `Handler`.

```suggestion
type GoRedis struct {
	Conn *goredis.Client // import goredis "github.com/go-redis/redis/v8"
	Ctx context.Context
}
```

```go
func (h *Handler) SetContext(ctx context.Context) *Handler
```

Reply:
Tagging @nitishm for suggestions